src/utils.ts

```diff
@@ -35,27 +35,85 @@ export type afterDOMUpdateHandle = {
   raf: number;
   timeout?: ReturnType<typeof setTimeout>;
 };
+
+const uncompletedDOMUpdates: Set<afterDOMUpdateHandle> = new Set();
+const afterAllDOMUpdateCallbacks: (() => void)[] = [];
+
+function markDOMUpdateCompleted(handle: afterDOMUpdateHandle) {
+  // remove the handle from the set of uncompleted handles
+  uncompletedDOMUpdates.delete(handle);
+
+  // check that there are still 0 updates left
+  // after each callback is called, since the callback could
+  // theoreticaly add more dom updates into the
+  // queue.
+  while (
+    uncompletedDOMUpdates.size === 0 &&
+    afterAllDOMUpdateCallbacks.length > 0
+  ) {
+    afterAllDOMUpdateCallbacks.shift()();
+  }
+}
+
+/**
+ * A combination of setTimeout and requestAnimationFrame that executes
+ * a callback function after a new animation frame is available and
+ * the given delay has past.
+ *
+ * @param callback callback to call
```

Comment:
Accidental doublespeak, or intentional?

Reply:
almost. the first "callback" is the name of the parameter. The second "callback" is the first word in the description for what that parameter is. Blame javadoc, their the ones who started it.

It looks more sensible in vscode:

![Screenshot from 2021-09-24 18-54-13](https://user-images.githubusercontent.com/124827/134754042-ddb3e5cf-7235-48db-891a-5e4ea89f727b.png)